### PR TITLE
Add a file for scratch-blocks-specific util functions

### PR DESCRIPTION
**Pull request by rachel-fenichel**
### Resolves

None
### Proposed Changes

Moves four functions into a new file + namespace, Blockly.scratchBlocksUtils.  The functions were previously in Blockly.utils.

### Reason for Changes

There are some utility functions that Blockly doesn't need but Scratch Blocks does.  This formalizes where they go, removes some potential merge conflicts, and makes it clearer when code is implementing scratch-blocks-specific behaviour.

### Test Coverage

Rebuilt and played around in the vertical playground.

### CODE REVIEW

File: core/field_textinput.js [hunk 1]
```diff
@@ -124,13 +126,15 @@ Blockly.FieldTextInput.prototype.init = function() {
...
   // If not in a shadow block, draw a box.
   if (notInShadow) {
-    this.box_ = Blockly.utils.createSvgElement('rect', {
-      'x': 0,
-      'y': 0,
-      'width': this.size_.width,
-      'height': this.size_.height,
-      'fill': this.sourceBlock_.getColourTertiary()
-    });
+    this.box_ = Blockly.utils.createSvgElement('rect',
```
Comment (rachel-fenichel) [inline core/field_textinput.js:129]: Drive-by lint, which has nothing else to do with this change.

**Review by kchadha — APPROVED** [review]
LGTM
